packages/dds/tree/src/simple-tree/api/schemaFactory.ts

```diff
@@ -331,15 +335,20 @@ export class SchemaFactory<
 	 * @privateRemarks
 	 * See note on array.
 	 */
-	public map<const T extends TreeNodeSchema | readonly TreeNodeSchema[]>(
+	public map<
+		const T extends TreeNodeSchema | readonly TreeNodeSchema[],
+		const TCustomMetadata = unknown,
```

Comment:
Unless I missed something, this seems impossible to define a good behavior for, see my comment on the structurally named array overload.